ament_cmake_core/cmake/environment_hooks/ament_cmake_environment_hooks_package_hook.cmake

```diff
@@ -1,7 +1,13 @@
+if(WIN32)
+    set(_ext .bat)
+else()
+    set(_ext .sh)
+endif()
 ament_environment_hooks(
-  "${ament_cmake_core_DIR}/environment_hooks/environment/ament_prefix_path.sh"
-  "${ament_cmake_core_DIR}/environment_hooks/environment/path.sh"
+  "${ament_cmake_core_DIR}/environment_hooks/environment/ament_prefix_path${_ext}"
+  "${ament_cmake_core_DIR}/environment_hooks/environment/path${_ext}"
 )
+unset(_ext)
```

Comment:
The underscore is used to prevent collisions with "reasonable" variables. You don't have to unset variables prefixed with an underscore.